### DISCUSSION TITLE
App not deploying?

### DISCUSSION THREAD

**Discussion by motykadaw** [vercel/community#602] [Help] [answered]
Hey, I have a node+express app and I managed to create a serverless vercel deployment for it. 
Well, almost. 
It works most of the time and at the last part for each function the logs say:
```
(... build stuff ...)
09:10:18.626 | Uploading build outputs...
09:10:19.053 | Deploying build outputs...
(... build stuff ...)
```
Unfortunately sometimes, when I push new commit to github, the build goes rather normally and the build logs match the logs from the correct deployments except this part, which just skips the `Deploying build outputs` part for some reason.
At this moment, the serverless functions do not work and behave like a static asset (POSTs to `x-app/api/function` get CORS error and GETs return the built js code.

Crucial points to notice:
- old deployed environments are still working
- redeploying the same or any previously working old commit (promoting to production) does the same thing from this moment forward
- reverting the branch to previously working commit does the same thing
- after unknown amount of time (usually trying the next day) the situation fixes itself and deploying the same code that was first to fail works fine.

I'm really frustrated by this behavior as I can't find any connection between what I'm doing and when that happens. Is there an undocumented caching feature that causes the deploy step to be skipped? Please help me - what am I missing here?

EDIT:
I'm at the "broken" phase and I've created a new project reading from the same repo. Still broken, so it should not be a caching problem.
EDIT 2:
A day has passed and the mysterious `Deploying build outputs...` appeared again on redeploying the latest commit yet again.

**Comment by swarnava** [comment, marked as answer]
Hi there, we had pushed a fix related to this yesterday. Apologise for confusion and I'm glad to hear things are working as expected now.
